source/PrintData.cpp

```diff
@@ -0,0 +1,724 @@
+
+	for(auto &it : GameData::Outfits())
+	{
+		// Skip non-weapons and submunitions.
+		if(!it.second.IsWeapon() || it.second.Category().empty())
+			continue;
+
+		const Outfit &outfit = it.second;
+		cout << it.first << ',';
+		cout << outfit.Category() << ',';
+		cout << outfit.Cost() << ',';
+		cout << -outfit.Get("weapon capacity") << ',';
+
+		cout << outfit.Range() << ',';
+
+		double reload = outfit.Reload();
+		cout << reload << ',';
+		cout << outfit.BurstCount() << ',';
+		cout << outfit.BurstReload() << ',';
+		cout << outfit.TotalLifetime() << ',';
+		double fireRate = 60. / reload;
+		cout << fireRate << ',';
+
+		double firingEnergy = outfit.FiringEnergy();
+		cout << firingEnergy << ',';
+		firingEnergy *= fireRate;
+		double firingHeat = outfit.FiringHeat();
+		cout << firingHeat << ',';
+		firingHeat *= fireRate;
+		double firingForce = outfit.FiringForce();
+		cout << firingForce << ',';
+		firingForce *= fireRate;
+
+		cout << firingEnergy << ',';
+		cout << firingHeat << ',';
+		cout << firingForce << ',';
+
+		double shieldDmg = outfit.ShieldDamage() * fireRate;
+		cout << shieldDmg << ',';
+		double dischargeDmg = outfit.DischargeDamage() * 100. * fireRate;
+		cout << dischargeDmg << ',';
+		double hullDmg = outfit.HullDamage() * fireRate;
+		cout << hullDmg << ',';
+		double corrosionDmg = outfit.CorrosionDamage() * 100. * fireRate;
+		cout << corrosionDmg << ',';
+		double heatDmg = outfit.HeatDamage() * fireRate;
+		cout << heatDmg << ',';
+		double burnDmg = outfit.BurnDamage() * 100. * fireRate;
+		cout << burnDmg << ',';
+		double energyDmg = outfit.EnergyDamage() * fireRate;
+		cout << energyDmg << ',';
+		double ionDmg = outfit.IonDamage() * 100. * fireRate;
+		cout << ionDmg << ',';
+		double slowDmg = outfit.SlowingDamage() * fireRate;
+		cout << slowDmg << ',';
+		double disruptDmg = outfit.DisruptionDamage() * fireRate;
+		cout << disruptDmg << ',';
+		cout << outfit.Piercing() << ',';
+		double fuelDmg = outfit.FuelDamage() * fireRate;
+		cout << fuelDmg << ',';
+		double leakDmg = outfit.LeakDamage() * 100. * fireRate;
+		cout << leakDmg << ',';
+		double hitforce = outfit.HitForce() * fireRate;
+		cout << hitforce << ',';
+
+		cout << outfit.Homing() << ',';
+		double strength = outfit.MissileStrength() + outfit.AntiMissile();
+		cout << strength << ',';
+
+		double damage = outfit.ShieldDamage() + outfit.HullDamage();
+		double deterrence = .12 * damage / outfit.Reload();
+		cout << deterrence << '\n';
+	}
+	cout.flush();
+}
+
+
+
+void PrintData::PrintEngineStats()
+{
+	cout << "name" << '\t' << "cost" << '\t' << "mass" << '\t' << "outfit space" << '\t'
+		<< "engine capacity" << '\t' << "thrust/s" << '\t' << "thrust energy/s" << '\t'
+		<< "thrust heat/s" << '\t' << "turn/s" << '\t' << "turn energy/s" << '\t'
+		<< "turn heat/s" << '\t' << "reverse thrust/s" << '\t' << "reverse energy/s" << '\t'
+		<< "reverse heat/s" << '\n';
+	for(auto &it : GameData::Outfits())
+	{
+		// Ship non-engines.
```

Comment:
```suggestion
		// Skip non-engines.
```
You mean this, right?